src/components/LoggedInHeader/index.js

```diff
@@ -0,0 +1,71 @@
+// @flow
+import React from 'react';
+import { Link } from 'react-router-dom';
+import { setIsLoggedIn, setUserInfo } from 'pages/SignIn/actions';
+import type { Action } from 'pages/SignIn/actions';
+import { connect } from 'react-redux';
+import firebase from 'config';
+import * as ROUTES from 'constants/routes';
+
+const mapDispatchToProps = {
+  setIsLoggedIn,
+  setUserInfo,
+};
+
+type Props = {|
+  userInfo: UserInfo,
+  setIsLoggedIn: (isLoggedIn: boolean) => Action,
+  setUserInfo: (userInfo: ?UserInfo) => Action,
+|};
+
+const styles = {
+  box: {
+    display: 'flex',
+    justifyContent: 'space-between',
+  },
+  logout: {
+    backgroundColor: '#c3e6bc',
+    border: '2px solid darkgreen',
+    borderRadius: '2px',
+    color: 'darkgreen',
+    fontSize: '20px',
+    fontWeight: 'bold',
+    margin: '10px',
+    padding: '5px',
+  },
+  text: {
+    color: 'darkgreen',
+    fontSize: '25px',
+    fontStyle: 'italic',
+  },
+  title: {
+    alignItems: 'center',
+    color: 'darkgreen',
+    display: 'flex',
+    justifyContent: 'center',
+    textDecoration: 'none',
+  },
+};
+
+const LoggedInHeader = (props: Props) => {
+  const { userInfo } = props;
+  const email = userInfo ? userInfo.email : null;
+  const logOutMessage = 'Log Out';
+  const logOut = async () => {
+    await firebase.auth().signOut();
+    props.setIsLoggedIn(false);
+    props.setUserInfo(null);
```

Comment:
we can merge setIsLoggedIn and setUserInfo, meaning
if setUserInfo is null then loggedIn is false, vice versa